payment-circle/src/main/java/org/stellar/anchor/paymentservice/circle/CirclePaymentService.java

```diff
@@ -65,14 +65,13 @@ private HttpClient getWebClient(boolean authenticated) {
   private <T> Mono<T> handleCircleError(HttpClientResponse response, ByteBufMono bodyBytesMono) {
     return bodyBytesMono
         .asString()
-        .flatMap(
+        .map(
```

Comment:
❓ Can you explain when `.flatMap()` or `.map()` is appropriate?

As I understood it, `Mono.flatMap()` was the correct usage, otherwise you should use `Object.map()`?

Reply:
In reactor objects that inherit from `Publisher`, both map and flatMap are called after the async part of the publisher finishes, but while `.map` is executed synchronously, `.flatMap` spins off a new Publisher whose execution will be async ([ref]).

[ref]: https://www.baeldung.com/java-reactor-map-flatmap